python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -248,7 +248,7 @@ def startup(self):
         self.sessionlock.startup()
         # Load the list of files, this time be verbose and print out a summary
         # of errors
-        self.update_index(verbose=True, firstRun=True)
+        self.update_index(True, True)
```

Comment:
@mesmith75 A lot of the work in this PR was either to try and keep memory usage as small as possible or to reduce a lot of different entries found by kcachegrind when profiling the  under certain tasks which were showing to take a while.
(There is an extremely minor overhead to using the attribute names here due to (un)packing dictionaries which was showing up in kcachegrind and I was trying to explore what was causing some methods to be called an obscene number of times. This does normally only exhibit when doing tests of calling the same method of the order of 100k+ times.)
I was seeing a large number of dictionary lookups which, whilst not bad (because the code is highly efficient), was still such a large number as to be concerning. (of the order of 1M calls to lookup something during constructing ~1k objects)

Exploring why some methods (such as a dict lookup) were being called a lot eventually lead to finding a highly nested trace just for assigning a default attribute which I was able to fix within the Proxy layer which accounts for (I'm guessing) ~90% of the perceived user slowness when dealing with large numbers of `GangaObject` type objects.

I suppose there are arguments for/against micro-optimisations but I prefer to try and keep the total number of calls to a minimum if required within a kcachegrind profile as it at least makes some performance profiles easier to read (imo).

Ultimately the change here has a negligible performance impact so if the code is clearer without this I'm happy to revert, but it's better to work without the argument names where possible I'd argue.